rlbot_gui/story/story_challenge_setup.py

```diff
@@ -0,0 +1,462 @@
+from datetime import datetime
+from os import path
+from typing import List, Tuple
+
+import json
+import random
+import time
+import traceback
+
+from rlbot.utils.game_state_util import GameState
+from rlbot.utils.structures.game_data_struct import GameTickPacket, Vector3
+from rlbot.parsing.match_settings_config_parser import (
+    game_mode_types,
+    boost_amount_mutator_types,
+    map_types,
+    max_score_types,
+    rumble_mutator_types,
+    match_length_types,
+)
+from rlbot.matchconfig.match_config import (
+    PlayerConfig,
+    MatchConfig,
+    MutatorConfig,
+    Team,
+)
+from rlbot.setup_manager import SetupManager
+from rlbot_gui import gui as rlbot_gui  # TODO: Need to remove circular import
+
+from rlbot_gui.story.load_story_descriptions import BOTS_CONFIG
+
+WITNESS_ID = random.randint(0, 1e5)
+
+DEBUG_MODE_SHORT_GAMES = False
+
+def setup_failure_freeplay(setup_manager: SetupManager, message: str, color_key="red"):
+    match_config = MatchConfig()
+    match_config.game_mode = game_mode_types[0]
```

Comment:
For readability, it may be preferable to just use the string. We have lots of reasons to never change them, so it should be reliable. Up to you though.

Reply:
I think I am going to keep this as is. I am almost certainly going to mistype things.